lib/queue.rb

```diff
@@ -1,31 +1,67 @@
 class Queue
-
+  
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(30)
+    # we don't have a front and back yet so they're both -1
+    @front = @back = -1
+    
   end
-
+  
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    # if queue is empty
+    if @front == -1 && @back == -1
+      @front = 0
+      # because there will be one element after we enqueue the first one
+      @back = 0
+    end
+    
+    # if queue is full
+    if ((@back + 1) % @store.length) == @front
+      raise ArgumentError.new("Error: queue is full")
+    end
+    
+    @store[@back] = element
+    # Do the modulo so that you won't go over the end of the array
+    @back = (@back + 1) % @store.length
   end
-
+  
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    # when queue is empty
+    if @front == @back
+      raise ArgumentError.new("Error: queue is empty")
+    end
```

Comment:
The Queue could be empty (both front and back are -1).

```suggestion
    if @front == @back && @back != -1
      raise ArgumentError.new("Error: queue is empty")
    end
```